presto-main/src/main/java/com/facebook/presto/sql/planner/plan/ExchangeNode.java

```diff
@@ -57,7 +55,18 @@
     public enum Scope
     {
         LOCAL,
-        REMOTE
+        REMOTE,
+        REMOTE_MATERIALIZED;
+
+        public boolean isLocal()
+        {
+            return this == LOCAL;
+        }
+
+        public boolean isRemote()
```

Comment:
What about using enum class with constructor ? (Similar to `BufferState` class)?

Reply:
Update: I see, `JoinDistributionType` use this style, we can keep this for now since I don't have strong opinion :)

Reply:
I think passing into the constructor is the way to go here